core/src/main/java/org/pentaho/platform/scheduler2/action/SchedulerOutputPathResolver.java

```diff
@@ -116,42 +117,53 @@ public void setActionUser( String actionUser ) {
   @Override
   public String resolveOutputFilePath() throws SchedulerException {
 
-    final String fileNamePattern = getFilename();
-    final String outputFilePath = getDirectory();
-
+    // IMPROVEMENT: This does not belong here. Move to the caller.
```

Comment:
could consider prefacing with TODO so that this is highlighted in editors and, I think, sonar